pkgs/development/compilers/factor-lang/wrapper.nix

```
@@ -0,0 +1,146 @@
+{ lib
+, stdenv
+, makeWrapper
+, factor-unwrapped
+, guiSupport ? true
+, cairo
+, freealut
+, gdk-pixbuf
+, glib
+, gnome2
+, gtk2-x11
+, libGL
+, libGLU
+, librsvg
+, graphviz
+, coreutils
+, gnugrep
+, libogg
+, libvorbis
+, openal
+, openssl
+, pango
+, pcre
+, udis86
+, zlib
+  # Libraries added to ld.so.cache
+, runtimeLibs ? [ ]
+  # Packages added to path (and ld.so.cache)
+, binPackages ? [ ]
+  # Extra vocabularies added to out/lib/factor
+, extraVocabs ? [ ]
+  # Enable default libs and bins to run most of the standard library code.
+, enableDefaults ? true
+}:
+let
+  inherit (lib) optional optionals optionalString;
+  # missing from lib/strings
+  escapeNixString = s: lib.escape [ "$" ] (builtins.toJSON s);
+  toFactorArgs = x: lib.concatStringsSep " " (map escapeNixString x);
+  defaultLibs = optionals enableDefaults (
+    [ libogg libvorbis openal openssl pcre udis86 zlib ] ++
+    (optionals guiSupport [
+      cairo
+      freealut
+      gdk-pixbuf
+      glib
+      gnome2.gtkglext
+      gtk2-x11
+      libGL
+      libGLU
+      pango
+    ]));
+  defaultBins = optionals enableDefaults [ graphviz ];
+  libs = defaultLibs ++ runtimeLibs ++ binPackages;
+  bins = binPackages ++ defaultBins;
+
+in
+stdenv.mkDerivation (finalArgs: {
```

```suggestion
stdenv.mkDerivation (finalAttrs: {
```
that's the standard name